gnat2goto/driver/driver.adb

```diff
@@ -254,16 +256,23 @@ package body Driver is
    function Is_Back_End_Switch (Switch : String) return Boolean is
       First : constant Natural := Switch'First + 1;
       Last  : constant Natural := Switch_Last (Switch);
-
+      use GNAT2GOTO.Options;
    begin
       --  For now we allow the -g/-O/-f/-m/-W/-w and -pipe switches, even
       --  though they will have no effect. This permits compatibility with
       --  existing scripts.
-
-      return
-        Is_Switch (Switch)
-          and then (Switch (First) in 'f' | 'g' | 'm' | 'O' | 'W' | 'w'
-                      or else Switch (First .. Last) = "pipe");
+      if Is_Switch (Switch) then
+         if Switch (First) in 'f' | 'g' | 'm' | 'O' | 'W' | 'w'
+           or else Switch (First .. Last) = "pipe"
+         then
+            return True;
+         elsif Switch (First .. Last) = No_Dump_Statement_AST_On_Error_Option
+         then
+            Dump_Statement_AST_On_Error := False;
```

Comment:
This function is called `Is_Back_End_Switch` so I was assuming originally this was only meant to be a check... but I couldn't find another place where we're passed any options, so I handled it here. If anyone has a better idea where to put this that'd be appreciated.

I also don't like setting this as a global option, but I don't really see how to pass options around otherwise as these are all free functions with a fixed interface (I can make it read-only elsewhere though with a bit of shuffling around of code)